mitodl/mit-open
# Remove LearningResourcePlatform.audience field, add LearningResource.is_professional field

# What are the relevant tickets?
Closes #97
Based on PR #99 (which should be reviewed and merged first)

# Description (What does it do?)
- Gets rid of the `LearningResourcePlatform.audience` field
- Adds a `LearningResource.is_professional` boolean field
- Fixes everything that broke as a result


# How can this be tested?
- Run `docker compose up` or run migrations via mgmt command
- Run `docker compose run --rm web ./manage.py backpopulate_xpro_data`
- Check that all the xpro course/program `LearningResource` objects now have `is_professional=True`
- Go to `http://localhost:8063/learning_resources/?professional=true`, you should get the xpro courses and/or programs returned
- Go to `http://localhost:8063/learning_resources/?professional=false`, you should get no courses/programs, just  podcasts/episodes returned if you previously ran `backpopulate_podcast_data`, or nothing at all otherwise.

# Additional Context
Whether a course/program is professional or not currently depends on which `LearningResourceOfferor` it is related to, so the new field could have been added to `LearningResourceOfferor` instead.  But I decided to add it to `LearningResource` so that additional criteria (or exceptions) could be accounted for in the future.
